api/tests/integration/edge_api/identities/test_edge_identity_featurestates_view.py

```diff
@@ -392,6 +494,65 @@ def test_edge_identities_update_featurestate(
     assert args[0] == expected_identity_document
 
 
+def test_edge_identities_with_identifier_update_featurestate(
+    admin_client,
+    environment,
+    environment_api_key,
+    identity_document,
+    dynamo_wrapper_mock,
+    feature,
+):
+    # Given
+    dynamo_wrapper_mock.get_item.return_value = identity_document
+    identifier = identity_document["identifier"]
+    featurestate_uuid = identity_document["identity_features"][0]["featurestate_uuid"]
+    url = reverse(
+        "api-v1:environments:edge-identities-with-identifier-featurestates-detail",
+        args=[environment_api_key, featurestate_uuid],
+    )
+    expected_feature_state_value = "new_feature_state_value"
+    expected_fs_enabled = True
+    data = {
+        "multivariate_feature_state_values": [],
+        "enabled": expected_fs_enabled,
+        "feature": feature,
+        "featurestate_uuid": featurestate_uuid,
+        "feature_state_value": expected_feature_state_value,
+        "identifier": identifier,
+    }
+
+    # When
+    response = admin_client.put(
+        url, data=json.dumps(data), content_type="application/json"
+    )
+    # Then
+    assert response.status_code == status.HTTP_200_OK
+    assert response.json()["feature"] == feature
+    assert response.json()["feature_state_value"] == expected_feature_state_value
+    assert response.json()["enabled"] == data["enabled"]
+    dynamo_wrapper_mock.get_item.assert_called_with(
+        f"{environment_api_key}_{identifier}"
+    )
+    name, args, _ = dynamo_wrapper_mock.mock_calls[1]
+    assert name == "put_item"
+
+    # Next, let's verify that the document that we put
+    # have correct updates
+
+    # First, let's create the copy of the original document
+    expected_identity_document = identity_document
+    # Next, let's modify the fs value that we updated
+    expected_identity_document["identity_features"][0][
+        "feature_state_value"
+    ] = expected_feature_state_value
+
+    # Next, let's update the enabled
+    expected_identity_document["identity_features"][0]["enabled"] = expected_fs_enabled
+
+    # Finally, let's compare them
+    assert args[0] == expected_identity_document
```

Comment:
As above, I feel like it would be more readable if we had a series of assert statements rather than building a copy of the document to compare.